### PR TITLE
refs #14259 - reduce data sent back from products index

### PR DESCRIPTION
**Pull request by komidore64**

### CODE REVIEW

File: app/views/katello/api/v2/products/index.json.rabl [hunk 1]
```diff
@@ -3,5 +3,20 @@ object false
...
+  attributes :sync_summary
+
+  child :sync_plan do
+    attributes :name, :interval
+  end
+
+  node :repository_count do |product|
+    product.library_repositories.count
+  end
 end
```
Comment (jlsherrill) [inline app/views/katello/api/v2/products/index.json.rabl]: The current convention is to put everything in base.json.rabl that index needs and share that with show.json.rabl.

Any reason to go against that convention?  

I'd rather see stuff moved from base.json.rabl to show.json.rabl to speed up index rendering.

Comment (komidore64) [inline app/views/katello/api/v2/products/index.json.rabl]: Ah okay. Not intending to go against any particular convention. Shoulda looked at the ``show`` template also before refactoring.

Reworking.

File: app/views/katello/api/v2/products/show.json.rabl [hunk 2]
```diff
@@ -21,6 +23,31 @@ child :provider do
   attribute :name
 end
 
+attributes :sync_plan_id
+
+child :sync_plan do
+  attributes :description, :sync_date, :next_sync
+end
```
Comment (jlsherrill) [inline app/views/katello/api/v2/products/show.json.rabl]: rabl sadly doesn't merge things together here.  So you'll need to put :name and :interval here too.

File: app/views/katello/api/v2/products/show.json.rabl [hunk 3]
```diff
@@ -3,16 +3,18 @@ object @resource
...
 
 attributes :available_content => :available_content
 
+attributes :redhat? => :redhat
+
 child :library_repositories => :repositories do |_repo|
   attributes :name, :id
 end
 
+attributes :available_content => :available_content, :if => params[:include_available_content]
```
Comment (jlsherrill) [inline app/views/katello/api/v2/products/show.json.rabl]: This is actually used as part of the activation key products page.  

https://github.com/Katello/katello/blob/master/engines/bastion_katello/app/assets/javascripts/bastion_katello/activation-keys/details/activation-key-products.controller.js#L37

File: app/views/katello/api/v2/products/show.json.rabl (hunk 2 above)
Comment (komidore64) [inline app/views/katello/api/v2/products/show.json.rabl]: haha yeah, i figured that out the hard way! I'll update it real soon.

File: app/views/katello/api/v2/products/show.json.rabl (hunk 3 above)
Comment (komidore64) [inline app/views/katello/api/v2/products/show.json.rabl]: Ah okay, thanks for pointing that out. I"ll add this back into ``base.json.rabl``.

**Comment by komidore64** [comment]
rebased

**Comment by beav** [comment]
PR may need another rebase to pick up test fixes

**Comment by komidore64** [comment]
rebased

**Comment by jlsherrill** [comment]
ACK

**Pull request merged**